pandas/core/generic.py

```diff
@@ -3218,14 +3219,22 @@ def filter(self, items=None, like=None, regex=None, axis=None):
                 **{name: [r for r in items if r in labels]})
         elif like:
             def f(x):
-                if not isinstance(x, string_types):
+                if isinstance(x, binary_type):
+                    x = bytes_to_str(x)
```

Comment:
hmm can you codify these into a function in ``pandas/compat/__init__.py``, maybe ``to_str``, and add a doc-string so its clear what it does, put it right below the ``u/u_safe`` functions

Reply:
make it clear that you are turning bytes, non-strings into strings